### PR TITLE
Align handling colors with RN

### LINKED ISSUE #5746: `useAnimatedStyle` does not support `hsl(x y% z%)` syntax for colors
### Description

When using `hsl(x y% z%)` syntax for background color in animated style, the color is not updated.

### Steps to reproduce

```tsx
import { StyleSheet, View } from 'react-native';

import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withRepeat,
  withTiming,
} from 'react-native-reanimated';
import React, { useEffect } from 'react';

export default function EmptyExample() {
  const sv = useSharedValue(0);

  useEffect(() => {
    sv.value = withRepeat(withTiming(1, { duration: 1000 }), -1, true);
    return () => {
      sv.value = 0;
    };
  }, [sv]);

  const animatedStyle = useAnimatedStyle(() => {
    return {
      backgroundColor: `hsl(${sv.value * 180} 100% 50%)`,
    };
  });

  return (
    <View style={styles.container}>
      <Animated.View style={[styles.box, animatedStyle]} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  box: {
    width: 100,
    height: 100,
  },
});
```

### Snack or a link to a repository

nope

### Reanimated version

3.7.1

### React Native version

0.73.4

### Platforms

Android, iOS

### JavaScript runtime

Hermes

### Workflow

React Native

### Architecture

Paper (Old Architecture)

### Build type

Debug app & dev bundle

### Device

iOS simulator

### Device model

iPhone 15 Pro

### Acknowledgements

Yes

### PR DESCRIPTION
**Pull request by maciekstosio**
## Summary

Fixes #5746. Not all color formats and syntax supported by RN is supported by reanimated. This PR should align with RN implementation. An example for rgb,rgba,hsl,hsl with comma and without (for view and Animated.View):

|Before|After|
|-|-|
| <video src="https://github.com/software-mansion/react-native-reanimated/assets/11800297/fe6f9b1d-a4e3-4542-baae-9d499aa6f252" />|<video src="https://github.com/software-mansion/react-native-reanimated/assets/11800297/1c33d1d3-6e0b-483a-a88d-7c8eb467b859" />|

## Test plan

Reused [RN tests](https://github.com/facebook/react-native/blob/main/packages/normalize-color/__tests__/normalizeColor-test.js).

<details>
<summary>Example from the video</summary>

```jsx
import { StyleSheet, View, Text } from 'react-native';

import Animated, {
  useAnimatedStyle,
  useSharedValue,
  withRepeat,
  withTiming,
} from 'react-native-reanimated';
import React, { useEffect } from 'react';

const CompareColumnAnimated = ({
  title,
  withComma,
}: {
  title: string;
  withComma?: boolean;
}) => {
  const sv = useSharedValue(0);

  const toRgb = (value: number) => {
    'worklet';
    return Math.floor(Math.abs(value) * 255);
  };

  useEffect(() => {
    sv.value = withRepeat(withTiming(1, { duration: 1000 }), -1, true);
    return () => {
      sv.value = 0;
    };
  }, [sv]);

  return (
    <View style={{}}>
      <View style={styles.header}>
        <Text>{title}</Text>
      </View>
      <Animated.View
        style={[
          styles.box,
          useAnimatedStyle(() => ({
            backgroundColor: `rgb(${toRgb(sv.value)} 255 255)`
              .split(' ')
              .join(withComma ? ', ' : ' '),
          })),
        ]}
      />
      <Animated.View
        style={[
          styles.box,
          useAnimatedStyle(() => ({
            backgroundColor: `rgba(${toRgb(sv.value)} 255 255 0.5)`
              .split(' ')
              .join(withComma ? ', ' : ' '),
          })),
        ]}
      />
      <Animated.View
        style={[
          styles.box,
          useAnimatedStyle(() => ({
            backgroundColor: `hsl(${sv.value * 180} 100% 50%)`
              .split(' ')
              .join(withComma ? ', ' : ' '),
          })),
        ]}
      />
      <Animated.View
        style={[
          styles.box,
          useAnimatedStyle(() => ({
            backgroundColor: `hsla(${sv.value * 180} 100% 50% 0.5)`
              .split(' ')
              .join(withComma ? ', ' : ' '),
          })),
        ]}
      />
    </View>
  );
};

const CompareColumn = ({
  title,
  withComma,
}: {
  title: string;
  withComma?: boolean;
}) => (
  <View style={{}}>
    <View style={styles.header}>
      <Text>{title}</Text>
    </View>
    <View
      style={[
        styles.box,
        {
          backgroundColor: 'rgb(100 255 255)'
            .split(' ')
            .join(withComma ? ', ' : ' '),
        },
      ]}
    />
    <View
      style={[
        styles.box,
        {
          backgroundColor: 'rgba(100 255 255 0.5)'
            .split(' ')
            .join(withComma ? ', ' : ' '),
        },
      ]}
    />
    <View
      style={[
        styles.box,
        {
          backgroundColor: 'hsl(100 100% 50%)'
            .split(' ')
            .join(withComma ? ', ' : ' '),
        },
      ]}
    />
    <View
      style={[
        styles.box,
        {
          backgroundColor: 'hsla(100 100% 50% 0.5)'
            .split(' ')
            .join(withComma ? ', ' : ' '),
        },
      ]}
    />
  </View>
);

export default function EmptyExample() {
  return (
    <View style={styles.container}>
      <CompareColumnAnimated title="anim comma" withComma />
      <CompareColumn title="view comma" withComma />
      <CompareColumnAnimated title="anim no comma" />
      <CompareColumn title="view no comma" />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
  },
  box: {
    width: 100,
    height: 100,
  },
  header: { justifyContent: 'center', alignSelf: 'center' },
});

```

</details>

### CODE REVIEW

File: src/reanimated2/Colors.ts [hunk 1]
```diff
@@ -297,7 +341,7 @@ export const ColorProperties = makeShareable([
   'overlayColor',
 ]);
 
-function normalizeColor(color: unknown): number | null {
+export function normalizeColor(color: unknown): number | null {
```
Comment (piaskowyk) [inline src/reanimated2/Colors.ts:344]: Do we need `export` here?

File: src/reanimated2/__tests__/Colors.test.ts
Comment (piaskowyk) [inline src/reanimated2/__tests__/Colors.test.ts:1]: Great tests! Let's just add test for short float notation - `.5`, for example: 'rgba(0, 0, 0, .5)`

**Review by piaskowyk — APPROVED** [review]

File: src/reanimated2/Colors.ts (hunk 1 above)
Comment (maciekstosio) [reply src/reanimated2/Colors.ts:344]: Yeah, we need to import it in tests

File: src/reanimated2/__tests__/Colors.test.ts
Comment (maciekstosio) [reply src/reanimated2/__tests__/Colors.test.ts:1]: Added

**Pull request merged**